falcon/request.py

```diff
@@ -1886,12 +1886,17 @@ class RequestOptions:
             any non-percent-encoded commas (default ``False``). When ``False``,
             values containing commas are left as-is. In this mode, list items
             are taken only from multiples of the same parameter name within the
-            query string (i.e. ``/?t=1,2,3&t=4`` becomes ``['1,2,3', '4']``).
+            query string (i.e. ``t=1,2,3&t=4`` becomes ``['1,2,3', '4']``).
             When `auto_parse_qs_csv` is set to ``True``, the query string value
             is also split on non-percent-encoded commas and these items
-            are added to the final list (i.e. ``/?t=1,2,3&t=4``
+            are added to the final list (i.e. ``t=1,2,3&t=4``
             becomes ``['1', '2', '3', '4']``).
 
+            Warning:
+                When `auto_parse_qs_csv` is set to either ``True`` or ``False``,
```

Comment:
I'm probably totally misunderstanding this line, but could you clarify what you mean with
> When `auto_parse_qs_csv` is set to either ``True`` or ``False``

As this option is designated as boolean, I expect it to be always set to either ``True`` or ``False`` :thinking:

Reply:
Yes exactly! What I'm saying is it doesn't matter what `auto_parse_qs_csv` is set to (either True or False), but either way this warning applies. Does that make sense?

Reply:
Maybe it's just me, but no, admittedly I'm still not following :imp:.

If `auto_parse_qs_csv` is set to the default (`False`) value, how does the following apply? :arrow_lower_left: 
> only the first set of values for the same key are are split.

To my understanding, if `auto_parse_qs_csv` is set to `False`, no values are split (at least in this context).

Reply:
agree

Reply:
I was able to confirm by adding a unit test that this only occurs when the option is enabled.